### PR TITLE
add array of enums casting for attributes with an array of enum values

### LINKED ISSUE #38: model attribute cast array of enums
We have a multiselect/checkboxes - every of these options is a value of the same enum.

```php
foreach($enum->values() as $value) {
    echo '<checkbox value="'.$value.'"/>';
}
```

And on the model we save this in a JSON column. like we already introduced the `:nullable` flag - how about being able to add a `array` flag or even better use the `[]` notation behind the enum class name.

Because the new Laravel 7 object custom casts work pretty much the same like our current notation we should/could add this feature before dropping Laravel 6 support.

```
MyEnum::class
MyEnum::class.':nullable'
// options for new array cast  feature
MyEnum::class.'[]:nullable'
MyEnum::class.':nullable,array'
```

The expected result would be an array of enum instances `Enumerable[]`.

### PR DESCRIPTION
**Pull request by Gummibeer**
closes #38

### CODE REVIEW

File: src/HasEnums.php [hunk 1]
```diff
@@ -132,6 +132,17 @@ protected function setEnumAttribute(string $key, $value)
             return $this;
         }
 
+        if ($this->isArrayOfEnums($key)) {
+            if (! is_array($value)) {
+                $modelClass = static::class;
+                throw new InvalidArgumentException("{$modelClass} [{$key}] expects array of enum values.");
```
Comment (brendt) [inline src/HasEnums.php:138]: In these cases I'd prefer to use a dedicated exception class, what do you think?

**Comment by brendt** [comment]
Ah how I miss generics :p Something like this would be so much nicer:

```php
protected $enums = [
    'status' => Collection<StatusEnum>,
];
```

Anyways, I think it's a great addition! I had one minor remark :)

**Comment by Gummibeer** [comment]
Generics would be harder to parse. Because we already had the `nullable` flag I think that's okay.
It's also only a solution for this last release before L7 only and the new custom casts.

File: src/HasEnums.php (hunk 1 above)
Comment (Gummibeer) [reply src/HasEnums.php:138]: Yeah. Good point. Will add one also for the nullable exception.

**Comment by brendt** [comment]
Good for me! Feel free to merge

**Pull request merged**